test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -95,3 +518,246 @@ func cpuSpecToString(cpus *performancev2.CPU) string {
 	}
 	return sb.String()
 }
+
+func createMachineConfig(profile *performancev2.PerformanceProfile) (*machineconfigv1.MachineConfig, error) {
```

Comment:
Since it's a tests code which used only once here, maybe it's simpler to read a complete MC manifest using `embed` and just apply it instead?